### PR TITLE
161142356 stop sequence in map

### PR DESCRIPTION
**Pull request by rckopteri**
# Fixed
* Stop sequence visualisation in map

### CODE REVIEW

File: ote/src/clj/ote/services/transit_visualization.clj [hunk 1]
```diff
@@ -18,32 +18,42 @@
...
-                              :coordinates [(Double/parseDouble lon)
-                                            (Double/parseDouble lat)]
-                              :properties {"name" name
-                                           "trip-name" (str (:name first-stop) " \u2192 " (:name last-stop))}})))
-                    (when-not (str/blank? stops)
-                      (str/split stops #"\|\|"))))))
-          trips))
+(defn trip-lines [trips]
+  (mapv
+    #(first (second %))
```
Comment (philosotractor) [inline ote/src/clj/ote/services/transit_visualization.clj]: Would this be a good place to comment what is expected to be got?

File: ote/src/clj/ote/services/transit_visualization.clj [hunk 2]
```diff
@@ -18,32 +18,42 @@
...
-                           (let [[lon lat name] (str/split stop #",")]
-                             {:type "Point"
-                              :coordinates [(Double/parseDouble lon)
-                                            (Double/parseDouble lat)]
-                              :properties {"name" name
-                                           "trip-name" (str (:name first-stop) " \u2192 " (:name last-stop))}})))
-                    (when-not (str/blank? stops)
-                      (str/split stops #"\|\|"))))))
-          trips))
+(defn trip-lines [trips]
```
Comment (philosotractor) [inline ote/src/clj/ote/services/transit_visualization.clj]: Food for thought: Would the long function with nested iterations become more readable, by e.g. splitting the loops into functions or naming the anonymous functions?

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 3]
```diff
@@ -689,85 +699,110 @@
...
-                       :checked show?
-                       :on-check #(e! (tv/->ToggleShowRouteLine routename))}])))]
-   [selected-route-map e! date->hash hash->color compare]])
+                              :margin-bottom "1.25rem"
+                              :display "flex"
+                              :flex-direction "row-reverse"
+                              :flex-wrap "nowrap"
+                              :justify-content "space-between"})
+
+     [:div {:style {:width "20%"}}
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: Was it tested if the :width percentages work on narrow screens like mobile?

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 4]
```diff
@@ -689,85 +699,110 @@
...
+      [ui/checkbox {:label "Näytä pysäkit"
+                    :checked (boolean (:show-stops? compare))
+                    :on-check #(e! (tv/->ToggleRouteDisplayStops))}]]
+     [:div {:style {:width "80%"}}
+      (when (> (count (:show-route-lines compare)) 0)
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: `pos-int?`

File: ote/src/clj/ote/services/transit_visualization.clj (hunk 2 above)
Comment (rckopteri) [reply ote/src/clj/ote/services/transit_visualization.clj]: You are correct. This is kind of hard to understand what is happening here and why. I try to add comments

File: ote/src/cljs/ote/views/transit_visualization.cljs [hunk 5]
```diff
@@ -689,85 +699,110 @@
...
-                       :on-check #(e! (tv/->ToggleShowRouteLine routename))}])))]
-   [selected-route-map e! date->hash hash->color compare]])
+                              :margin-bottom "1.25rem"
+                              :display "flex"
+                              :flex-direction "row-reverse"
+                              :flex-wrap "nowrap"
+                              :justify-content "space-between"})
+
+     [:div {:style {:width "20%"}}
+      [ui/checkbox {:label "Näytä pysäkit"
```
Comment (philosotractor) [inline ote/src/cljs/ote/views/transit_visualization.cljs]: l8n, a.k.a localization :)

File: ote/src/clj/ote/services/transit_visualization.clj (hunk 1 above)
Comment (rckopteri) [reply ote/src/clj/ote/services/transit_visualization.clj]: added

File: ote/src/cljs/ote/views/transit_visualization.cljs (hunk 4 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/transit_visualization.cljs]: Changed

File: ote/src/cljs/ote/views/transit_visualization.cljs (hunk 3 above)
Comment (rckopteri) [reply ote/src/cljs/ote/views/transit_visualization.cljs]: I didn't test with mobile screens. This page won't work in narrow screens, but I did take flex into use in this. Thank you for pointing this out.

**Pull request merged**